apps/design/backend/src/app.ts

```diff
@@ -985,6 +1006,15 @@ export function buildUnauthenticatedApi({ logger, workspace }: AppContext) {
           pageIndex,
         } = decodeQuickResultsMessage(payload);
 
+        // Get the exported election data and validate it
+        const exportedElectionResult =
+          await store.getExportedElection(ballotHash);
+        if (exportedElectionResult.isErr()) {
+          return exportedElectionResult;
+        }
+        const election = exportedElectionResult.ok();
+
+        // Also get the election record for metadata
```

Comment:
It makes me a little nervous that all of these endpoints load both the database election record and also the exported election. Seems like it would be easy to get them mixed up. Is it possible to just use the exported election? E.g. we could have a store method `getLastExportedElection(electionId)`